### PR TITLE
Update Time logic to be aligned with configuration + update config (#154/#171)

### PR DESCRIPTION
**Pull request by xiaodoudou**
Adjust time logic calculation to align configuration and solve reported bug #154

### CODE REVIEW

**Comment by xiaodoudou** [comment]
Updated the config notes for #171

**Comment by xiaodoudou** [comment]
People are complaining sometimes they cannot sleep.

I'm suspecting the following check to be in cause:

```cs
            if (!ZNet.instance.IsServer())
            {
                Debug.Log("Is not server.");
                return;
            }
```

My assumption is maybe the client sync with the server and both have their time sync periodically. 
What put be on that is the following report we got on discord:
```
Dat Rigby ThoToday at 9:58 AM
Looks like this might be a bug. Are you doing this on a dedi? I went and changed my client settings to match my server settings and now it's actually working. It seems like the server knows it's still "day" which is why I couldn't sleep until deep into the night. I went and changed my client settings and logged back in, and now it's daytime in the server instead of night.
Seems like there's some desync between the time of day on server and what the client settings are expecting.
Guess I'll be leaving it vanilla until that's fixed.
```

Maybe what we are having is a local client seen a night time when it's a day time on the server?
A test of 3 people could be a confirmation for it or more debugging and checking what the game is doing could confirm this.

**Comment by xiaodoudou** [comment]
Just to link to that PR with bug #170; better cover the all in one.
I am running some tests related to the `isServer` condition to see if that what's troubling sleeping to be triggered.

**Comment by xiaodoudou** [comment]
Here the result of testing:

- Environment: Dedicated Linux server with V+ 9.0.0 + Windows client on V+ 9.0.0
- Unusual behavior: Was able to sleep during day time
- The condition `Is not server.` is triggering only when a sleeping phase on the client side
- Configuration is:

```
[Time]

; Change false to true to enable this section
enabled=true

; Total amount of time one complete day and night circle takes to complete
totalDayTimeInSeconds=1800

; Increase the speed at which time passes at night by %. The value 50 would result in a 50% reduced amount of night time.
nightTimeSpeedMultiplier=50
```

**Comment by xiaodoudou** [comment]
After some debug on client/server; seems the `Boolean isNight = __instance.IsNight();` not in phase with the sleeping feature.
Was able to sleep when the server was returning `isNight at false`

And after got weird desync on time when the client was lock in `isNight at true` and server was always considering its day time `isNight  at false`.

Trying to see how we can work around that.

**Comment by xiaodoudou** [comment]
After checking Valheim code, It seems we are able to sleep during the afternoon:
```cs
[...]
else if (this.IsMine())
	{
		ZLog.Log("Is mine");
		if (this.IsCurrent())
		{
			ZLog.Log("is current spawnpoint");
			if (!EnvMan.instance.IsAfternoon() && !EnvMan.instance.IsNight())
			{
				human.Message(MessageHud.MessageType.Center, "$msg_cantsleep", 0, null);
				return false;
			}
[...]
```

Adding the time-skip check seems to solve the time desync.

**Comment by xiaodoudou** [comment]
Ready for review, but that approach seems to works after my testing:

- Only the server is emitting netTime
- Avoid speed time during a time skip (Sleeping/Bedtime)
- Ensure we don't overlap the morning start (0.25% of a day)

I have tested it for a very short period and was able to see stuff working properly, note the ability to sleep during the afternoon is confusing at first, but it is intended by Valheim vanilla.

**Comment by xiaodoudou** [comment]
forget to push my lastest commit.

File: ValheimPlus/RPC/VPlusConfigSync.cs [hunk 1]
```diff
@@ -79,6 +79,9 @@ public static void RPC_VPlusConfigSync(long sender, ZPackage configPkg)
 
                             Configuration.Current = ConfigurationExtra.LoadFromIni(memStream);
 
+                            // Needed to make sure client is using server configuration as dayLength is setup before
+                            TimeManipulation.SetupDayLength();
```
Comment (xiaodoudou) [inline ValheimPlus/RPC/VPlusConfigSync.cs:83]: @Mixone-FinallyHere not sure if I should put that here; but to highlight the issue; I have spotted if the client has different dayLength it is set up before network exchange, but doing such call here realign it.

**Comment by xiaodoudou** [comment]
Merge time changes with 0.9.2.

If someone could give a test to see if it works fine and didn't forget anything that would make 2 bugs issues solved.

**Comment by radmint** [comment]
@xiaodoudou could you merge master in so that all of the recent changes aren't showing as "Files changed" in the PR to make it a bit easier to read? Thank you!

**Comment by xiaodoudou** [comment]
@radmint done ;-)

File: valheim_plus.cfg [hunk 2]
```diff
@@ -110,7 +110,8 @@ fermenterItemsProduced=6
 ; If changed to enabled all fireplaces do not need to be refilled.
 enabled=false
 
-; If you enable this option only placed torches do not need to be refilled.
+; If you enable this option with the previous option to true, only placed torches do not need to be refilled. 
+; Apply to: wood torches, iron torches, green torches, sconces and brazier
```
Comment (radmint) [inline valheim_plus.cfg:114]: thank you for this update!

**Review by nxPublic — APPROVED** [review]

**Comment by nxPublic** [comment]
This does look like the correct solution, we will make sure to test and debug this before public release.

Awesome work!

❤️

**Pull request merged**